### DISCUSSION TITLE
any tips for rendering 4096x4096 pngs in a resource-constrained setting?

### DISCUSSION THREAD

**Discussion by hn-88** [OpenSpace/OpenSpace#3028] [Q&A] [answered]
Hello OpenSpace community,

Thank you for providing this software which gives very beautiful imagery. After hearing about OpenSpace for a few years now, I've finally got hold of an NVidia graphics card and was able to try it out. 

I'm located in India and our internet connection is not very reliable - I get only around 10 Mbps while downloading textures and other assets. Additionally, the monitor I'm using for content creation doesn't support modes higher than 1920x1080. With these constraints, I'd like to render 4K (4096x4096) png image sequences of some impressive content like this flight through the digital universe, https://www.youtube.com/watch?v=dju9yRz5k6A

So I have some questions.

1. Is the osrec file for flight through the digital universe (and other scenes useful for planetariums) available somewhere? I did not find it on [[OpenSpace Hub](http://hub.openspaceproject.com).](http://hub.openspaceproject.com/)

2. With Stellarium, I had worked out that for getting hi-res imagery, one way would be to use vncserver -geometry 7680x4320 on the machine running the software, and view the screen from another machine (or the same machine) with Tightvnc viewer, choosing 25% zoom in options - this was some years ago when Stellarium didn't yet offer screenshots of arbitrary resolution. Now Stellarium does have the feature of screenshots of arbitrary resolution, and the code is available on Stellarium's repo. Would it be possible for someone on the Openspace community to implement something similar for the "Output frames" option? That is, have an additional "output width" x "output height" option instead of defaulting to the screen resolution? Or is there some other more streamlined solution for generating 4096x4096 fisheye (fulldome) pngs?

3. Is there some setting using which we can save the cache/textures/tiles or make the cache "permanent" for some profile(s) so that once the download is done, the next time we open that profile or that osrec file, OpenSpace doesn't download the cache again?

Thank you for your time.

**Comment by GPayne** [comment, marked as answer]
Welcome, and thanks for your interest!

Your low download speed will mean that it takes longer to sync at the beginning, but once it's loaded there won't really be any further delay.

To your questions:
1. You will need to create your own recording of flying through the digital universe, since I don't think anyone else has provided one. This [wiki page](https://docs.openspaceproject.com/en/latest/manual/content/session-recording.html) explains how to start/stop a recording session.

2. OpenSpace already has this ability for rendering images that are higher resolution than the live display. For example, look at the file config/single.json. The window entry has a "size" of 1280x720, but a "res" of 2560x1440. This means that OpenSpace will run in a 1280x720 window, but if you save a screenshot or play-back a session recording with the "output frames" option, you will get a .png image for each frame at 2560x1440 resolution. So for your purposes you could copy single.json and create your own config file with "res" of 4096x4096.

3. Look into slack Q&A on this topic.
Go into the openspace.cfg file and set `ModuleConfigurations->GlobeBrowsing->MRFCacheEnabled` to true. Run OpenSpace again it will cache the globebrowsing data for planet/moon locations you visit. Then later, it should use the cached data for those same locations.
There is also an Offline profile that you can select from the launcher, and see if that has the content you want.

**Reply by hn-88** [reply]
That's great! Thank you, @GPayne !
